Platon/backend/app/search_engine/views.py

```diff
@@ -203,6 +233,82 @@ def get(user_id,self):
         else:
             return make_response(jsonify({"error": "Missing data fields or invalid data."}), 400)
 
+@search_engine_ns.route("/workspace")
+class WorkspaceSearchAPI(Resource):
+
+    @api.doc(responses={401 : 'Account Problems', 400 : 'Input Format Error' ,500 : 'Database Connection Error'})
+    @api.response(200, 'Valid Search Result', workspace_list_model)
+    @api.expect(ws_search_parser)    
+    def get(self):
+        form = WorkspaceSearchForm(request.args)
+        if form.validate():
+            search_query = form.search_query.data.lower()
+            # Remove the punctuation in the search query
+            search_query = SearchEngine.remove_punctuation(search_query)
+            # Tokenize the search query
+            tokens = search_query.split()
+            # Get stopwords form the API
+            stopwords = SearchEngine.get_stopwords()
+            # Remove stopwords from the list of tokens
+            tokens = list(set(tokens) - set(stopwords))
+            # Get the semantically related list for our tokens
+            tokens = SearchEngine.semantic_related_list(tokens)
+            # List of user dictionaries that will be send as a search output
+            result_list = []
+            # Score of each user record
+            result_id_score = []
+            for token,score in tokens:
+                query = '(LOWER(title) REGEXP ".*{0}.*" OR LOWER(description) REGEXP ".*{0}.*")'.format(token)
+                sql_statement = "SELECT * FROM workspaces WHERE {}".format(query)
+                result = db.engine.execute(sql_statement)
+                for workspace in result:
+                    id_list = [i[0] for i in result_id_score]
+                    if workspace[0] not in id_list:
+                        # Remove private worksapces
+                        if workspace[2] == 1:
```

Comment:
Minor suggestion: instead of filtering out the private workspaces here, we can add "is_private=True" to the query above.